Billing history tab

hovex023 opened the pull request:
What:  To implement the Billing History Portion of the new Billing flow/redesign

Why: To create a better user experience

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

NikolaiYurchenko commented on web/satellite/src/components/account/billing/billingTabs/BillingHistorHeader.vue:
```diff
@@ -0,0 +1,153 @@
+        return this.sortBy === AccessGrantsOrderBy.NAME;
+    }
+
+    /**
```
wrong jsdoc position?

NikolaiYurchenko commented on web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue:
```diff
@@ -0,0 +1,57 @@
+        VList,
+        BillingHistoryHeader,
+        BillingHistoryShape,
+        VPagination
```
trailing comma

NikolaiYurchenko commented on web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue:
```diff
@@ -0,0 +1,57 @@
+
+<style scoped lang="scss">
+     .billing_history2 {
+            position: relative;
```
7 spaces, should be 4

NikolaiYurchenko commented on web/satellite/src/components/account/billing/billingTabs/Coupons.vue:
```diff
@@ -0,0 +1,27 @@
+}
+</script>
+
+<style scoped lang="scss">
```
no empty tags

NikolaiYurchenko reviewed (changes requested):
and resolve merge conflict

hovex023 commented on web/satellite/src/components/account/billing/billingTabs/Coupons.vue:
```diff
@@ -0,0 +1,27 @@
+import { RouteConfig } from '@/router';
+
+// @vue/component
+@Component({
```
<img width="756" alt="image" src="https://user-images.githubusercontent.com/97616907/176734266-243d3e4f-f8bb-4abf-adb2-5aa3784f5ab7.png">
Looks like linter wants it to be formatted that way

NikolaiYurchenko reviewed (dismissed):
generally LG

mobyvb commented on web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue:
```diff
@@ -1,22 +1,52 @@
+
 export default class BillingArea extends Vue {
+    public get historyItems(): PaymentsHistoryItem[] {
+        return this.$store.state.paymentsModule.paymentsHistorly;
```
typo "paymentsHistorly"

hovex023 replied:
fixed

mobyvb reviewed (commented):
looks good overall but I noticed a typo - I can approve when this is fixed
Another thing I noticed - you uploaded both svgs and pngs, but in the code, only the svgs are used. Are the pngs necessary?
I also saw you flipped one use of `isNewAccessGrantFlow` - was this intentional?

hovex023 commented:
> looks good overall but I noticed a typo - I can approve when this is fixed Another thing I noticed - you uploaded both svgs and pngs, but in the code, only the svgs are used. Are the pngs necessary? I also saw you flipped one use of `isNewAccessGrantFlow` - was this intentional?

Removed the typo.  Also removed the unnecessary png files. The linter was complaining about those flags for some reason. Looking at the access grant section, I believe the flags are correct right now

mobyvb commented on web/satellite/src/components/account/billing/billingTabs/BillingHistorHeader.vue:
```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2022 Storj Labs, Inc.
```
nit - typo in filename

mobyvb commented on web/satellite/src/components/account/billing/billingTabs/BillingHistoryShape.vue:
```diff
@@ -0,0 +1,228 @@
+        CheckIcon,
+    },
+})
+export default class BillingHistoryShape extends Vue {
```
Why is this component called BillingHistoryShape?

hovex023 replied:
Vlist takes a dom shape as a property. So I named it BillinghistoryShape as it's passed into Vlist.

mobyvb commented on web/satellite/src/components/accessGrants/AccessGrants.vue:
```diff
@@ -169,7 +169,7 @@
                 </span>
             </div>
         </div>
-        <div v-if="!isNewAccessGrantFlow">
```
why is this flag flipped? With this change, the old access flow no longer properly lists accesses. This is the only comment blocking my approval. Everything else looks good to me, and I'm happy to approve once the old access grants flow works on this branch

hovex023 replied:
fixed
